microsoft/devhome
# Introducing Windows Customization

## Summary of the pull request
This commit includes the initial placeholder L1 page for 'Windows customization' which will host various settings and utilities to customize Windows to a developer's preferences.

![image](https://github.com/microsoft/devhome/assets/7259210/07fdd650-9b76-4500-8e50-4c307b958b4c)

![image](https://github.com/microsoft/devhome/assets/7259210/5ffdf7fe-4721-407c-8a9f-1d53972b4d64)

## References and relevant issues
Starts the work related to the "Advanced Settings" feature exploration #1983 

## Detailed description of the pull request / Additional comments
The MainPage acts as a root which eventually will host a ContentPresenter for displaying the active view (for example, when search is added the ContentPresent will switch the root view temporarily to the 'All settings' search view, and once we get better support to integrate into the navigation service, sub-pages could be consolidated into that content presenter as well.

To provide an example of how options can be incorporated, the Developer File Explorer page was introduced.

The pattern we’re going for here is to provide logical settings groups in the form of a UserControl, which is then wrapped by a Page. This helps in the future when implementing search because we want to host all settings controls in the same root view and filter visibility based on the search filter. Having each group as a separate control view can easily enable hosting that UI in any form we need to down the line. The sub-pages themselves are probably temporarily placeholders for a better integration with the navigation service down the line.

See tools\Customization\DevHome.Customization\Customization.md for details.

## Validation steps performed
- Verified that the Windows customization section appears on the navigation menu and can be navigated to.
- Clicking on the Windows developer settings link will navigate to the Windows Settings app "For developers" page
- Toggle each setting on "Developer File Explorer" and verify that the setting takes effect, both on and off
- 
## PR checklist
- [x] Makes progress on #1983
